.github/workflows/codeql-analysis.yml

```diff
@@ -19,11 +19,11 @@ jobs:
         # a pull request then we can checkout the head.
         fetch-depth: 2
 
-    # If this run was triggered by a pull request event, then checkout
```

Comment:
This has nothing to do with the Java version but I noticed this warning in the build while checking this so I figured I would add it:

>  issue was detected with this workflow: git checkout HEAD^2 is no longer necessary. Please remove this step as Code Scanning recommends analyzing the merge commit for best results.